blueocean-dashboard/src/main/js/components/karaoke/rest/KaraokeApi.js

```diff
@@ -49,6 +50,27 @@ function parseNewStart(response) {
     return response;
 }
 
+function recursivelyFetchNodes(href, fetchOptions) {
```

Comment:
Would it be possible to add 1-2 unit tests for this method? You can see an example of how we've done some similar unit tests with mocks or the `nock` library in the `blueocean-core-js` module. Check out the specs for `Capabilities` and `Fetch`.